### PR TITLE
Improve coverage on Windows

### PR DESCRIPTION
**Pull request by Nikokrock**

### CODE REVIEW

File: e3/os/process.py [hunk 1]
```diff
@@ -687,15 +689,15 @@ def is_running(pid):
         handle = NT.OpenProcess(Access.PROCESS_QUERY_INFORMATION, False, pid)
 
         try:
-            if handle == 0:
+            if not handle:
                 return False
-
+            print(handle)
```
Comment (enzbang) [inline e3/os/process.py]: I guess that was for debug only?

File: e3/os/windows/fs.py [hunk 2]
```diff
@@ -321,7 +321,11 @@ def iterate_on_dir(self, fun, default_result=None):
                                        pointer(self.io_status),
                                        b, b_size, FileInfo.Names.class_id,
                                        False, None, True)
-        if status == Status.NO_MORE_FILES:
+        if status == Status.NO_MORE_FILES:  # defensive code
+            # In theory this case should not occurs at it means that the
+            # directory does not even have the . and .. entries. In practice
+            # it can occurs (probably because of an intermediate states).
```
Comment (enzbang) [inline e3/os/windows/fs.py]: state?

File: tests/tests_e3/os/windows/fs/main_test.py [hunk 3]
```diff
@@ -192,6 +207,16 @@ def fun(name, ntfile_instance):
     finally:
         ntfile.close()
 
+    test_file = os.path.join(test_dir_path, 'not-a-directory.txt')
+    touch(test_file)
+    try:
+        ntfile = NTFile(test_file)
```
Comment (enzbang) [inline tests/tests_e3/os/windows/fs/main_test.py]: try: should start after ntfile

File: tests/tests_e3/os/windows/fs/main_test.py [hunk 4]
```diff
@@ -273,3 +298,43 @@ def test_unlink():
...
     ntfile.close()
     assert 'NTFile.read_attributes:' in str(err)
+
+    # A directory that is not empty cannot be deleted
+    dir_to_delete = os.path.join(test_dir_path, 'dir_to_delete')
+    mkdir(dir_to_delete)
+    touch(os.path.join(dir_to_delete, 'afile.txt'))
+
+    try:
+        ntfile = NTFile(dir_to_delete)
```
Comment (enzbang) [inline tests/tests_e3/os/windows/fs/main_test.py]: try after assignment

File: tests/tests_e3/os/windows/fs/main_test.py [hunk 5]
```diff
@@ -273,3 +298,43 @@ def test_unlink():
...
+
+    # A directory that is already opened and not empty cannot be
+    # moved to trash
+    dir_to_delete = os.path.join(test_dir_path, 'dir_to_delete')
+    mkdir(dir_to_delete)
+    touch(os.path.join(dir_to_delete, 'afile.txt'))
+
+    try:
+        ntfile = NTFile(dir_to_delete)
+        ntfile2 = NTFile(dir_to_delete)
```
Comment (enzbang) [inline tests/tests_e3/os/windows/fs/main_test.py]: try after assignment

File: tests/tests_e3/os/windows/fs/main_test.py [hunk 6]
```diff
@@ -273,3 +298,43 @@ def test_unlink():
...
+    finally:
+        ntfile.close()
+        ntfile2.close()
+
+    # Try to delete a file that we cannot open
+    try:
+        touch(deleted_file_path)
+        ntfile = NTFile(deleted_file_path)
+        ntfile.open(Access.READ_DATA, Share.NOTHING)
+        ntfile2 = NTFile(deleted_file_path)
```
Comment (enzbang) [inline tests/tests_e3/os/windows/fs/main_test.py]: same here for try:

File: tests/tests_e3/os/windows/fs/main_test.py (hunk 3 above)
Comment (enzbang) [reply tests/tests_e3/os/windows/fs/main_test.py]: you can probably use with closing(...):

File: tox.ini [hunk 7]
```diff
@@ -4,7 +4,6 @@ envlist = py27-cov
 [testenv]
 deps =
      pytest
-     pytest-catchlog
```
Comment (enzbang) [inline tox.ini]: I would prefer a separated commit for this one ;)

File: e3/os/process.py (hunk 1 above)
Comment (Nikokrock) [reply e3/os/process.py]: right will remove

File: e3/os/windows/fs.py (hunk 2 above)
Comment (Nikokrock) [reply e3/os/windows/fs.py]: correct

**Pull request merged**